ydb/core/scheme/scheme_tablecell.h

```diff
@@ -10,79 +10,148 @@
-        float FloatVal;
-        char Bytes[8];
+        ui64 Size : 62;
+        ui64 Kind : 2;
+    };
+
+    struct TInlineValue {
+        char Data[15];
+        ui8 Size : 6;
+        ui8 Kind : 2;
+    };
+
+private:
+    union {
+        char Raw[16];
+        TRefValue Ref;
+        TInlineValue Inline;
     };
 
+    constexpr ui8 GetKind() const noexcept {
+        return ui8(Raw[15]) >> 6;
+    }
+
+    constexpr bool IsRefValue() const noexcept {
+        return (GetKind() & 2) == 0;
+    }
+
+    constexpr bool HasInlineFlag() const noexcept {
+        return (GetKind() & 1) == 0;
+    }
+
 public:
     TCell()
         : TCell(nullptr, 0)
     {}
 
     TCell(TArrayRef<const char> ref)
-        : TCell(ref.begin(), ui32(ref.size()))
-    {
-        Y_ABORT_UNLESS(ref.size() < Max<ui32>(), " Too large blob size for TCell");
-    }
-
-    TCell(const char* ptr, ui32 size)
-        : DataSize_(size)
-        , IsInline_(0)
-        , IsNull_(ptr == nullptr)
-        , Ptr(ptr)
-    {
-        Y_DEBUG_ABORT_UNLESS(ptr || size == 0);
-
-        if (CanInline(size)) {
-            IsInline_ = 1;
-            IntVal = 0;
+        : TCell(ref.data(), ref.size())
+    {}
 
+    TCell(const char* ptr, size_t size) {
+        if (!ptr) {
+            Y_DEBUG_ABORT_UNLESS(size == 0);
+            // All zeroes represents the null value
+            ::memset(Raw, 0, 16);
+        } else if (CanInline(size)) {
+            // We use switch with constant size memcpy for better codegen
             switch (size) {
-                case 8: memcpy(&IntVal, ptr, 8); break;
-                case 7: memcpy(&IntVal, ptr, 7); break;
-                case 6: memcpy(&IntVal, ptr, 6); break;
-                case 5: memcpy(&IntVal, ptr, 5); break;
-                case 4: memcpy(&IntVal, ptr, 4); break;
-                case 3: memcpy(&IntVal, ptr, 3); break;
-                case 2: memcpy(&IntVal, ptr, 2); break;
-                case 1: memcpy(&IntVal, ptr, 1); break;
+                case 15: ::memcpy(Inline.Data, ptr, 15); break;
+                case 14: ::memcpy(Inline.Data, ptr, 14); break;
+                case 13: ::memcpy(Inline.Data, ptr, 13); break;
+                case 12: ::memcpy(Inline.Data, ptr, 12); break;
+                case 11: ::memcpy(Inline.Data, ptr, 11); break;
+                case 10: ::memcpy(Inline.Data, ptr, 10); break;
+                case 9: ::memcpy(Inline.Data, ptr, 9); break;
+                case 8: ::memcpy(Inline.Data, ptr, 8); break;
+                case 7: ::memcpy(Inline.Data, ptr, 7); break;
+                case 6: ::memcpy(Inline.Data, ptr, 6); break;
+                case 5: ::memcpy(Inline.Data, ptr, 5); break;
+                case 4: ::memcpy(Inline.Data, ptr, 4); break;
+                case 3: ::memcpy(Inline.Data, ptr, 3); break;
+                case 2: ::memcpy(Inline.Data, ptr, 2); break;
+                case 1: ::memcpy(Inline.Data, ptr, 1); break;
+                case 0: break;
```

Comment:
```suggestion
                case 0: break;
                default: std::unreachable();
```
```suggestion
                case 0: break;
                default: std::unreachable();
            }
```
Or any alternative, I think it should improve codegen

Reply:
It already covers all possible cases, no need for unreachable.